### PR TITLE
Change BeanDiscovery test to account for beans.xml version attribute …

### LINKED ISSUE #305: Adjust test(s) related to bean discovery in CDI Full
Changes done in https://github.com/eclipse-ee4j/cdi/pull/549 remove the affect of present/absent `version` attribute of beans.xml on chosen bean discovery mode.
I recall that I saw at least one test that verifies this. We should look that up and adjust it.

### PR DESCRIPTION
**Pull request by manovotn**
…changes. Expand BeansXml util to allow creation of XML without certain attributes.

Fixes #305

### CODE REVIEW

**Comment by manovotn** [comment]
Come to think of it, I am not sure why we have `web/src/main/java/org/jboss/cdi/tck/tests/deployment/discovery/integration/BeanDiscoveryTest.java` in place because there is also `org.jboss.cdi.tck.tests.full.deployment.discovery.BeanDiscoveryTest`.
It is either completely redundant or at least has some useless bits in it...

File: impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java [hunk 1]
```diff
@@ -239,9 +251,19 @@ public BeansXml setBeanDiscoveryMode(BeanDiscoveryMode mode) {
...
     public InputStream openStream() {
         StringBuilder xml = new StringBuilder();
-        xml.append("<beans version=\"" + getBeansXmlVersion().getValue() + "\" bean-discovery-mode=\"");
-        xml.append(getBeanDiscoveryMode().getValue());
-        xml.append("\">\n");
+        xml.append("<beans");
+        BeansXmlVersion version = getBeansXmlVersion();
+        // add version if required
+        if (version != null) {
+            xml.append(" version=\"" + getBeansXmlVersion().getValue() + "\"");
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java:258]: Redundant `getBeansXmlVersion()` call:

```suggestion
            xml.append(" version=\"" + version.getValue() + "\"");
```

File: impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java [hunk 2]
```diff
@@ -239,9 +251,19 @@ public BeansXml setBeanDiscoveryMode(BeanDiscoveryMode mode) {
...
+        }
+        BeanDiscoveryMode mode = getBeanDiscoveryMode();
+        // add bean discovery mode if required
+        if (mode != null) {
+            xml.append(" bean-discovery-mode=\"" + getBeanDiscoveryMode().getValue() + "\"");
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/shrinkwrap/impl/BeansXml.java:263]: Redundant `getBeanDiscoveryMode()` call:

```suggestion
            xml.append(" bean-discovery-mode=\"" + mode.getValue() + "\"");
```

**Review by Ladicek — APPROVED** [review]

**Comment by Ladicek** [comment]
I guess we already have a test for `beans.xml` for Lite? Can't remember :-)

**Comment by manovotn** [comment]
> I guess we already have a test for `beans.xml` for Lite? Can't remember :-)

We have `EmptyBeansXmlDiscoveryTest`. Otherwise I am not sure.

**Comment by manovotn** [comment]
Added a commit that merges together the two `BeanDiscoveryTest` classes that we had (one was in core TCK, the other under `web`) as they were both testing just CDI Full features anyway. I didn't manage to decipher why this test got split in the first place.

**Pull request merged**